hedgehog/src/Hedgehog/Internal/TH.hs

```diff
@@ -13,6 +14,9 @@ import           Hedgehog.Internal.Discovery
 import           Hedgehog.Internal.Property
 
 import           Language.Haskell.TH
+#if MIN_VERSION_template_haskell(2, 12, 0)
+                    hiding (TExpQ)
+#endif
 import           Language.Haskell.TH.Syntax
```

Comment:
I can make all the import qualified or explicit, either just for this file or for all files. Your choice.